benches/topology.rs

```diff
@@ -434,10 +434,357 @@ fn benchmark_complex(c: &mut Criterion) {
+                    ),
+                );
+                config.add_sink(
+                    "company_unfurler_sink",
+                    &["company_unfurler_filter"],
+                    sinks::socket::SocketSinkConfig::make_basic_tcp_config(
+                        out_addr_company_unfurler.to_string(),
+                    ),
+                );
+                config.add_sink(
+                    "audit_sink",
+                    &["audit_rename"],
+                    sinks::socket::SocketSinkConfig::make_basic_tcp_config(
+                        out_addr_audit.to_string(),
+                    ),
+                );
+
+                let mut rt = runtime();
+                let (
+                    output_lines_company_api,
+                    output_lines_company_admin,
+                    output_lines_company_media_proxy,
+                    output_lines_company_unfurler,
+                    output_lines_audit,
+                    topology,
+                ) = rt.block_on(async move {
+                    let output_lines_company_api =
+                        CountReceiver::receive_lines(out_addr_company_api);
+                    let output_lines_company_admin =
+                        CountReceiver::receive_lines(out_addr_company_admin);
+                    let output_lines_company_media_proxy =
+                        CountReceiver::receive_lines(out_addr_company_media_proxy);
+                    let output_lines_company_unfurler =
+                        CountReceiver::receive_lines(out_addr_company_unfurler);
+                    let output_lines_audit = CountReceiver::receive_lines(out_addr_audit);
+
+                    let (topology, _crash) = start_topology(config.build().unwrap(), false).await;
+                    wait_for_tcp(in_addr).await;
+                    (
+                        output_lines_company_api,
+                        output_lines_company_admin,
+                        output_lines_company_media_proxy,
+                        output_lines_company_unfurler,
+                        output_lines_audit,
+                        topology,
+                    )
+                });
+                (
+                    rt,
+                    topology,
+                    output_lines_company_api,
+                    output_lines_company_admin,
+                    output_lines_company_media_proxy,
+                    output_lines_company_unfurler,
+                    output_lines_audit,
+                )
+            },
+            |(
+                mut rt,
+                topology,
+                output_lines_company_api,
+                output_lines_company_admin,
+                output_lines_company_media_proxy,
+                output_lines_company_unfurler,
+                output_lines_audit,
+            )| {
+                rt.block_on(async move {
+                    // Generate the inputs.
+                    let lines = [
```

Comment:
Should we move the input generation up into the setup block?

Reply:
Done! The effect is not significant though - below the noise threshold.